### PR TITLE
NIFI-6305: RecordPath Standalone Functions: format() and toDate() allow time zone argument

### PR DESCRIPTION
**Pull request by archongum**
https://issues.apache.org/jira/browse/NIFI-6305
https://issues.apache.org/jira/browse/NIFI-6245

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by MikeThomsen** [comment]
Why not extend the existing functions?

**Comment by archongum** [comment]
@MikeThomsen Do you mean both `format( /date, "yyyy-MM-dd HH:mm:ss")` and `format( /date, "yyyy-MM-dd HH:mm:ss", "GMT+8:00")` functions can work together at the same time? It is possible to keep a standalone function which can accept 2 arguments or 3 arguments together? 
If it can I would like to change the code. 
If not, it's not a good idea to modify the existing 2-arguments `format()` function to make it only can accept 3 arguments because early version NiFI using the 2-arguments one.

**Comment by archongum** [comment]
@MikeThomsen Changed to extend the existing functions

File: nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Format.java [hunk 1]
```diff
@@ -57,14 +66,22 @@ public Format(final RecordPathSegment recordPath, final RecordPathSegment dateFo
...
             return null;
         }
 
         try {
-            return DataTypeUtils.getDateFormat(dateFormatString);
+            if (timeZoneID == null) {
+                return DataTypeUtils.getDateFormat(dateFormatString);
+            } else {
+                final String timeZoneStr = RecordPathUtils.getFirstStringValue(timeZoneID, context);
+                if (timeZoneStr == null || timeZoneStr.isEmpty()) {
```
Comment (MikeThomsen) [inline nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Format.java]: Should use `StringUtils.isEmpty(String)`

Comment (MikeThomsen) [inline nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Format.java]: Should use `StringUtils.isEmpty(String)`

Comment (archongum) [reply nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Format.java]: It seems there is no `commons-lang3` artifact in `nifi-record-path` module. 
FYI, the code-style of RecordPath Functions is `charsetString == null || charsetString.isEmpty()` and I just followed it.

Comment (archongum) [reply nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Format.java]: It seems there is no `commons-lang3` artifact in `nifi-record-path` module. 
FYI, the code-style of RecordPath Functions is `charsetString == null || charsetString.isEmpty()` and I just followed it.

Comment (MikeThomsen) [reply nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/functions/Format.java]: There's a NiFi version of `StringUtils` that is part of our API.

File: nifi-commons/nifi-record-path/pom.xml [hunk 2]
```diff
@@ -66,6 +66,11 @@
             <artifactId>nifi-record</artifactId>
             <version>1.10.0-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-properties</artifactId>
```
Comment (bbende) [inline nifi-commons/nifi-record-path/pom.xml]: Was this dependency actually needed for anything?

Comment (archongum) [reply nifi-commons/nifi-record-path/pom.xml]: `StringUtils.isEmpty()` is in this dependency. At my first commit, I used `str == null || str.isEmpty()` but Mike suggests to use `StringUtils.isEmpty()` instead and `nifi-record-path` hasn't, has it?

**Comment by MikeThomsen** [comment]
Merged. Thanks for the patch!

**Pull request closed without merging**